File: scripts/srv.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type Response struct {
	Message string `json:"message"`
}

func upstreamHandler(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "application/json")
	response := Response{Message: "Hello, World!"}
	json.NewEncoder(w).Encode(response)
}

func main() {
	http.HandleFunc("/json", upstreamHandler)
	fmt.Println("Listening on :8090")
	if err := http.ListenAndServe(":8090", nil); err != nil {
		panic(err)
	}
}

File: scripts/srv.mjs
import { createServer } from 'node:http';

const server = createServer((req, res) => {
    if (req.url === '/json') {
        res.setHeader('Content-Type', 'application/json');
        const response = { message: 'Hello, World!' };
        res.end(JSON.stringify(response) + '\n');
    } else {
        res.statusCode = 404;
        res.end('Not Found\n');
    }
});

server.listen(3000, () => {
    console.log('Listening on :3000');
});